homeassistant/data_entry_flow.py

```diff
@@ -216,9 +200,11 @@ def __init__(
         """Initialize the flow manager."""
         self.hass = hass
         self._preview: set[str] = set()
-        self._progress: dict[str, FlowHandler] = {}
-        self._handler_progress_index: dict[str, set[FlowHandler]] = {}
-        self._init_data_process_index: dict[type, set[FlowHandler]] = {}
+        self._progress: dict[str, BaseFlowHandler] = {}
+        self._handler_progress_index: dict[str, set[BaseFlowHandler]] = {}
+        self._init_data_process_index: dict[type, set[BaseFlowHandler]] = {}
+
+    _flow_result: Callable[..., _FlowResultT]
```

Comment:
Where did Marc's comment go?

Reply:
```suggestion
    _flow_result: type[_FlowResultT]
```

Reply:
And move it above the init method.

Reply:
> Where did Marc's comment go?

`type[_FlowResultT]` would be ideal / the correct annotation, however mypy does appear to have an issue with it. Will need to investigate that one. That's why I deleted my initial comment (also thought I had only staged it in the first place - well guess I pressed the wrong button 😅)

_Moving it above the `__init__` method does still make sense though._

Reply:
Thanks! 👍

Reply:
I already tried the `type` version, but mypy does not like it like Marc suggests:
```
homeassistant/data_entry_flow.py:495: error: Cannot instantiate type "Type[FlowResult]"  [misc]
```

We could add a comment explaining `type` would be preferred but is not allowed by mypy? Another alternative would be to type it as `type` and override the resulting errors. That's somewhat annoying though because child classes would then *also* need to silence errors if instantiating.

Reply:
`type[X]` and `Callable[..., X]` are basically equivalent in the return type. The only difference being that `type[X]` also checks the init args.

I think it's fine to leave `Callable` for now. Feel free to add a comment though that `type` would be preferred. We can adjust it if / when mypy is fixed.

Reply:
Found the mypy issue for it: https://github.com/python/mypy/issues/11644
And here is the fix: https://github.com/python/mypy/pull/16963. Should be in the release after the next one, i.e. `1.10`.